tools/generators/normalize-package-dependencies/index.ts

```diff
@@ -0,0 +1,198 @@
+  const projectDependencies = getProjectDependenciesFromGraph(projectConfig.name!, graph);
+  const packageJsonPath = joinPathFragments(projectConfig.root, 'package.json');
+
+  updateJson(tree, packageJsonPath, json => {
+    updateDepType(json, 'dependencies');
+    updateDepType(json, 'devDependencies');
+
+    return json;
+  });
+
+  return tree;
+
+  function updateDepType(json: PackageJson, depType: 'dependencies' | 'devDependencies' | 'peerDependencies') {
+    const deps = json[depType];
+    if (!deps) {
+      return;
+    }
+
+    for (const packageName in deps) {
+      if (isProjectDependencyAnWorkspaceProject(graph, packageName, projectDependencies)) {
+        deps[packageName] = '*';
+      }
+    }
+  }
+}
+
+function reportPackageJsonDependenciesIssues(issues: ProjectIssues) {
+  const issueEntries = Object.entries(issues);
+
+  if (issueEntries.length === 0) {
+    return;
+  }
+
+  issueEntries.forEach(([projectName, dependencyIssues]) => {
+    logger.log(chalk.bold(chalk.red(`${projectName} has following dependency version issues:`)));
+    // eslint-disable-next-line guard-for-in
+    for (const dep in dependencyIssues) {
+      logger.log(chalk.red(`  - ${dep}`));
+    }
+  });
+
+  logger.info(`All these dependencies version should be specified as '*'`);
+  logger.info(`Fix this by running 'nx workspace-generator normalize-package-dependencies'`);
+
+  throw new Error('package dependency violations found');
+}
+
+function getPackageJsonDependenciesIssues(
+  tree: Tree,
+  projectConfig: ProjectConfiguration,
+  graph: ProjectGraph,
+): Record<string, string> | null {
+  const projectDependencies = getProjectDependenciesFromGraph(projectConfig.name!, graph);
+  const packageJsonPath = joinPathFragments(projectConfig.root, 'package.json');
+  const packageJson = readJson<PackageJson>(tree, packageJsonPath);
+
+  let issues: Record<string, string> | null = null;
+  checkDepType(packageJson, 'dependencies');
+  checkDepType(packageJson, 'devDependencies');
+  checkDepType(packageJson, 'peerDependencies');
+
+  return issues;
+
+  function checkDepType(json: PackageJson, depType: 'dependencies' | 'devDependencies' | 'peerDependencies') {
+    const deps = json[depType];
+    if (!deps) {
+      return null;
+    }
+
+    for (const packageName in deps) {
+      if (isProjectDependencyAnWorkspaceProject(graph, packageName, projectDependencies) && deps[packageName] !== '*') {
+        issues = issues ?? {};
+        issues[packageName] = deps[packageName];
+      }
+    }
+
+    return issues;
+  }
+}
+
+function getProjectDependenciesFromGraph(projectName: string, graph: ProjectGraph): string[] {
+  const projectDeps = graph.dependencies[projectName];
+  return projectDeps.map(value => value.target);
+}
+
+function isProjectDependencyAnWorkspaceProject(
+  graph: ProjectGraph,
+  dependencyName: string,
+  projectDependenciesFromGraph: string[],
+) {
+  const projectDeps = graph.dependencies[dependencyName];
```

Comment:
I agree this is bit more "advance" behaviour. 


we need to use nx dep graph because it will tell us if a project dependency is from workspace or not.

for example in react-northstar, it has  `"@fluentui/dom-utilities": "^1.1.1",` as dependency specified.  if we checked just against `readProjectConfig('@fluentui/dom-utilities')` it would resolve to `true`, but that's wrong as in workspace we have `"@fluentui/dom-utilities": "^2.x",`.  Nx graph understands this and will define that `1.x` dep as `npm:"@fluentui/dom-utilities`, which means it is not part of workspace. 

does it make sense ?